tests/test_manifest.py

```diff
@@ -763,13 +767,11 @@ def test_create_manifests(
             assert all_results == expected_result
 
     @pytest.mark.parametrize(
-        "component,datasetId,expected_file_based,expected_rows,expected_files",
+        "component, datasetId, expected_rows, expected_files, expected_annotations",
         [
-            ("Biospecimen", "syn61260107", False, 4, None),
```

Comment:
`TestManifestGenerator.test_get_manifest_first_time` already tests record based manifest generation, so it didn't seem necessary to duplicate that here. However, for file based manifests it only tested the nested dataset case without the `contentType:dataset` annotation.
The checks in that test have been replicated here and test cases are included for the top-level folder structure and nested dataset structure. Other dataset structures, ie `contentType:dataset` annotation datasets with nested files, and datasets with doubly nested files, can be added in the future but are tested in `TestStoreSynapse.test_getFilesInStorageDataset`